File: flutter17_mobile/lib/models/phone.dart
import 'package:json_annotation/json_annotation.dart';

part 'phone.g.dart';

@JsonSerializable()
class Phone {
  int? id;
  int? productId;

  // Display
  String? screenSize;
  String? screenResolution;
  String? screenType;
  int? refreshRate;

  // Performance
  String? processor;
  int? ram;
  int? storageCapacity;
  bool? supportsExpandableStorage;

  // Camera
  int? rearCamerasCount;
  bool? hasUltrawideLens;
  bool? hasZoomLens;
  int? mainCameraResolution;
  int? frontCameraResolution;

  // Battery
  int? batteryCapacity;
  bool? supportsFastCharging;
  bool? supportsWirelessCharging;
  int? estimatedBatteryLife;

  // Connectivity
  bool? supports5G;
  bool? hasWiFi6;
  bool? hasBluetooth;
  bool? hasNFC;
  bool? hasDualSIM;

  // Physical Characteristics
  double? weight;
  String? dimensions;
  String? buildMaterial;

  // Additional Features
  String? operatingSystem;
  bool? hasFingerprintSensor;
  bool? hasFaceRecognition;
  bool? isWaterResistant;

  Phone({
    this.id,
    this.productId,
    this.screenSize,
    this.screenResolution,
    this.screenType,
    this.refreshRate,
    this.processor,
    this.ram,
    this.storageCapacity,
    this.supportsExpandableStorage,
    this.rearCamerasCount,
    this.hasUltrawideLens,
    this.hasZoomLens,
    this.mainCameraResolution,
    this.frontCameraResolution,
    this.batteryCapacity,
    this.supportsFastCharging,
    this.supportsWirelessCharging,
    this.estimatedBatteryLife,
    this.supports5G,
    this.hasWiFi6,
    this.hasBluetooth,
    this.hasNFC,
    this.hasDualSIM,
    this.weight,
    this.dimensions,
    this.buildMaterial,
    this.operatingSystem,
    this.hasFingerprintSensor,
    this.hasFaceRecognition,
    this.isWaterResistant,
  });

  factory Phone.fromJson(Map<String, dynamic> json) => _$PhoneFromJson(json);

  Map<String, dynamic> toJson() => _$PhoneToJson(this);
}

File: angular17_web/src/app/models/phone/phone.ts
export class Phone {
  id!: number;
  productId!: number;

  // Display
  screenSize!: string;
  screenResolution!: string;
  screenType!: string;
  refreshRate!: number;

  // Performance
  processor!: string;
  ram!: number;
  storageCapacity!: number;
  supportsExpandableStorage!: boolean;

  // Camera
  rearCamerasCount!: number;
  hasUltrawideLens!: boolean;
  hasZoomLens!: boolean;
  mainCameraResolution!: number;
  frontCameraResolution!: number;

  // Battery
  batteryCapacity!: number;
  supportsFastCharging!: boolean;
  supportsWirelessCharging!: boolean;
  estimatedBatteryLife!: number;

  // Connectivity
  supports5G!: boolean;
  hasWiFi6!: boolean;
  hasBluetooth!: boolean;
  hasNFC!: boolean;
  hasDualSIM!: boolean;

  // Physical Characteristics
  weight!: number;
  dimensions!: string;
  buildMaterial!: string;

  // Additional Features
  operatingSystem!: string;
  hasFingerprintSensor!: boolean;
  hasFaceRecognition!: boolean;
  isWaterResistant!: boolean;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.productId = data.productId;
    this.screenSize = data.screenSize;
    this.screenResolution = data.screenResolution;
    this.screenType = data.screenType;
    this.refreshRate = data.refreshRate;
    this.processor = data.processor;
    this.ram = data.ram;
    this.storageCapacity = data.storageCapacity;
    this.supportsExpandableStorage = data.supportsExpandableStorage;
    this.rearCamerasCount = data.rearCamerasCount;
    this.hasUltrawideLens = data.hasUltrawideLens;
    this.hasZoomLens = data.hasZoomLens;
    this.mainCameraResolution = data.mainCameraResolution;
    this.frontCameraResolution = data.frontCameraResolution;
    this.batteryCapacity = data.batteryCapacity;
    this.supportsFastCharging = data.supportsFastCharging;
    this.supportsWirelessCharging = data.supportsWirelessCharging;
    this.estimatedBatteryLife = data.estimatedBatteryLife;
    this.supports5G = data.supports5G;
    this.hasWiFi6 = data.hasWiFi6;
    this.hasBluetooth = data.hasBluetooth;
    this.hasNFC = data.hasNFC;
    this.hasDualSIM = data.hasDualSIM;
    this.weight = data.weight;
    this.dimensions = data.dimensions;
    this.buildMaterial = data.buildMaterial;
    this.operatingSystem = data.operatingSystem;
    this.hasFingerprintSensor = data.hasFingerprintSensor;
    this.hasFaceRecognition = data.hasFaceRecognition;
    this.isWaterResistant = data.isWaterResistant;
  }
}
